MetaMask/metamask-extension
# [e2e] Switch ethereum chain multiple dapps

## **Description**
_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

Closes https://github.com/MetaMask/MetaMask-planning/issues/1432
Closes https://github.com/MetaMask/MetaMask-planning/issues/911

E2E tests for switchEthereumChain for multiple dapps and notification queueing for those dapps.

## **Manual testing steps**

_1. Pull the branch down
_2. yarn && yarn build:test && yarn test:e2e:chrome:rpc
_3. Tests should pass

## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before**

_[screenshot]_

### **After**

_[screenshot]_

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
